### ISSUE TITLE
problem during configuration

### ISSUE THREAD

**Issue by Alessandro1983** [roleoroleo/4-noks_ha_integration#1]
Hi, I follow the instructions to install 4-noks Elios4you in Home Assistant. I put the ip address but appears that it is impossible to obtain the serial number. If it is helpfull I fund the ip address by the router.

**Comment by roleoroleo** [comment]
I will check it asap.

**Comment by roleoroleo** [comment]
Try to run this script and check the output.
Change the IP address according to your device.
```
import socket

host = '192.168.1.200'
port = 5001

def get_inf():
    """Get system info from device."""
    error = False

    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    try:
        s.connect((host, port))
    except socket.error:
        _LOGGER.error("Failed to connect to device %s", host)
        error = True
    except OSError:
        _LOGGER.error("Failed to connect to device %s", host)
        error = True

    if error:
        return None

    try:
        s.sendall(b'@inf\n')
        data = s.recv(2048)
        s.close()
    except socket.error:
        _LOGGER.error("Failed to get status from device %s", host)
        error = True
    except OSError:
        _LOGGER.error("Failed to get status from device %s", host)
        error = True

    if error:
        s.close()
        return None

    print(data.decode("utf-8"))

def get_dat():
    """Get data from device."""
    error = False

    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    try:
        s.connect((host, port))
    except socket.error:
        _LOGGER.error("Failed to connect to device %s", host)
        error = True
    except OSError:
        _LOGGER.error("Failed to connect to device %s", host)
        error = True

    if error:
        return None

    try:
        s.sendall(b'@dat\n')
        data = s.recv(2048)
        s.close()
    except socket.error:
        _LOGGER.error("Failed to get data from device %s", host)
        error = True
    except OSError:
        _LOGGER.error("Failed to get data from device %s", host)
        error = True

    if error:
        s.close()
        return None

    print(data.decode("utf-8"))

get_inf()
get_dat()
```

**Comment by Alessandro1983** [comment]
> Try to run this script and check the output. Change the IP address according to your device.
> 
> ```
> import socket
> 
> host = '192.168.1.200'
> port = 5001
> 
> def get_inf():
>     """Get system info from device."""
>     error = False
> 
>     s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>     try:
>         s.connect((host, port))
>     except socket.error:
>         _LOGGER.error("Failed to connect to device %s", host)
>         error = True
>     except OSError:
>         _LOGGER.error("Failed to connect to device %s", host)
>         error = True
> 
>     if error:
>         return None
> 
>     try:
>         s.sendall(b'@inf\n')
>         data = s.recv(2048)
>         s.close()
>     except socket.error:
>         _LOGGER.error("Failed to get status from device %s", host)
>         error = True
>     except OSError:
>         _LOGGER.error("Failed to get status from device %s", host)
>         error = True
> 
>     if error:
>         s.close()
>         return None
> 
>     print(data.decode("utf-8"))
> 
> def get_dat():
>     """Get data from device."""
>     error = False
> 
>     s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>     try:
>         s.connect((host, port))
>     except socket.error:
>         _LOGGER.error("Failed to connect to device %s", host)
>         error = True
>     except OSError:
>         _LOGGER.error("Failed to connect to device %s", host)
>         error = True
> 
>     if error:
>         return None
> 
>     try:
>         s.sendall(b'@dat\n')
>         data = s.recv(2048)
>         s.close()
>     except socket.error:
>         _LOGGER.error("Failed to get data from device %s", host)
>         error = True
>     except OSError:
>         _LOGGER.error("Failed to get data from device %s", host)
>         error = True
> 
>     if error:
>         s.close()
>         return None
> 
>     print(data.decode("utf-8"))
> 
> get_inf()
> get_dat()
> ```

run the script from where?

**Comment by roleoroleo** [comment]
Any computer on the same local network with Python installed.

**Comment by alexf981** [comment]
Hi, 
this integration would be very useful for me.
I have the same problem. I try to run the scritp and the output was empty.

**Comment by alexf981** [comment]
Hi,
I tried to force the serial number in the python code, the wizard create the integration, however no sensor or entity was created, I have an empty integration.
I ask wath kind of configuration do you have in elios4you if it is direct or indirect. I use indirect configuration (because I want to see data when I am not at home) and in Elios App in diagnostic, Elios IP isn't one IP of the LAN but external IP. The script in this thread seems can connect to Elios but like I sead @dat e @inf are empty.
in wath version of Elios du you have tested the intergation?

Thanks a lot 
Alex

**Comment by roleoroleo** [comment]
I created this component for a friend and now I don't have a device to test.
I don't know what is the configuration about direct and indirect, but I can post the responde to @inf command:
```
ready...
@inf
@INF
FWTOP=V1293build164
FWBTM=V1030build29
SN=xxxxxxxx
HWVER=01000002000100
BTVER=M00
HW WIFI=GS2011
S2W APP VERSION=5.8.1
S2W GEPS VERSION=5.8.1
S2W WLAN VERSION=5.8.1
```

**Comment by BlackSake** [comment]
i have the same problem, indirect configuration. the script output was empty

**Comment by roleoroleo** [comment]
Did you change the ip address in the second line?
The script is not able to autodetect your device.

I think that indirect configuration should work.
My friend use it.

If it doesn't work, probably there is a different fw version.
Are you able to capture the network traffic near your device?

**Comment by BlackSake** [comment]
yes, i changed the IP address and network traffic is ok, all works in the app too.

**Comment by roleoroleo** [comment]
Please, try to catpure network traffic when you use the app.
